python/cuml/random_projection/random_projection.pyx

```diff
@@ -197,12 +197,17 @@ cdef class BaseRandomProjection():
         self.params.n_samples = n_samples
         self.params.n_features = n_features
 
-        if self.dtype == np.float32:
-            RPROJfit[float](handle_[0], self.rand_matS, &self.params)
-        else:
-            RPROJfit[double](handle_[0], self.rand_matD, &self.params)
+        try:
+            if self.dtype == np.float32:
+                RPROJfit[float](handle_[0], self.rand_matS, &self.params)
+            else:
+                RPROJfit[double](handle_[0], self.rand_matD, &self.params)
 
-        self.handle.sync()
+            self.handle.sync()
+
+        except RuntimeError:
```

Comment:
Seems like we're just catching the exception in order to reraise it? Why not just pass the original RuntimeError through?

Reply:
This is a good point. This was segfaulting before so I added `except +` to the relevant externed cython functions and added the explicit `try/except`. 

I will test this without the `try/except` blocks to see if it still segfaults. One thing I would like to see sometime in the near future is for the C++ layer to throw our own custom exception types so that we can catch them in Python and display more reasonable messages. I believe we have a couple Github issues related to this already. I've been finding users get turned off by the formatting (and content) of the C++ errors.